heapsort/heapsort.py

```diff
@@ -0,0 +1,25 @@
+def s(l):
+    #size
+    n = len(l)
+    print("n:",n)
+    print("range:", range(n,-1,-1))
+    #build heap
+    for i in range(n,-1,-1):
+        max = i
+        l_idx = 2*i+1
+        r_idx = 2*i+2
+        print("max:",max)
+        print("l_idx:",l_idx)
+        print("r_idx:",r_idx)
+        if l_idx < n and l[i] < l[l_idx]:
+            max=l_idx
+        print("max -1:",max)
+        if r_idx < n and l[max] < l[r_idx]:
+            max=r_idx
+        print("max -2:",max)
+        if max!=i:
+            l[i],l[max] = l[max],l[i]
+        print("l:",l)
+    for i in range(n-1, 0, -1):
```

Comment:
For the actual sorting part once you build the heap, you only do the first step (swap root element of heap with current index of sorted list we are building). Once you do this step once, the heap is no longer a heap! You need to run the build heap routine again to fix your heap.